test/itf/ItfLiquidatorShouldClosePositionTest.ts

```diff
@@ -0,0 +1,1127 @@
+import hre from "hardhat";
+import chai from "chai";
+import { Signer, BigNumber } from "ethers";
+import { MockSpreadModel } from "../../types";
+import {
+    ZERO,
+    USD_28_000_18DEC,
+    USD_50_000_18DEC,
+    PERCENTAGE_3_18DEC,
+    PERIOD_25_DAYS_IN_SECONDS,
```

Comment:
> add test
> 
> when some swaps achieve maturity and the rest don't when some swaps pnl>100% or pnl < -100% and not achieve maturity

When some swaps achieve maturity and others don't, how can I force some swaps to reach maturity without explicitly calling the `itfCloseSwapPayFixed` or `itfCloseSwapReceiveFixed` function and adding 28 days to that?

I would do something like this
```
        const closeTimestamp = paramsPayFixed.openTimestamp.add(PERIOD_28_DAYS_IN_SECONDS);

        await miltonDai.connect(paramsPayFixed.from).itfCloseSwapPayFixed(3, closeTimestamp);
        await miltonDai.connect(paramsPayFixed.from).itfCloseSwapReceiveFixed(8, closeTimestamp);
```

But this would be the same as the test on line 297 👇
```
    it("should close 5 pay fixed, 5 receive fixed positions in one transaction - case 2, some of them are already closed", async () => {
        //given
        miltonSpreadModel.setCalculateQuotePayFixed(BigNumber.from("6").mul(N0__01_18DEC));
        const testData = await prepareTestData(
            BigNumber.from(Math.floor(Date.now() / 1000)),
            [admin, userOne, userTwo, userThree, liquidityProvider],
            ["DAI"],
            [PERCENTAGE_5_18DEC],
            miltonSpreadModel,
            MiltonUsdcCase.CASE3,
            MiltonUsdtCase.CASE3,
            MiltonDaiCase.CASE3,
            MockStanleyCase.CASE1,
            JosephUsdcMockCases.CASE0,
            JosephUsdtMockCases.CASE0,
            JosephDaiMockCases.CASE0
        );

        await prepareApproveForUsers(
            [userOne, userTwo, userThree, liquidityProvider],
            "DAI",
            testData
        );
        await setupTokenDaiInitialValuesForUsers(
            [admin, userOne, userTwo, userThree, liquidityProvider],
            testData
        );

        const { tokenDai, josephDai, iporOracle, miltonDai, miltonStorageDai } = testData;
        if (
            tokenDai === undefined ||
            josephDai === undefined ||
            miltonDai === undefined ||
            miltonStorageDai === undefined
        ) {
            expect(true).to.be.false;
            return;
        }

        const paramsPayFixed = getPayFixedDerivativeParamsDAICase1(userTwo, tokenDai);
        const paramsReceiveFixed = getReceiveFixedDerivativeParamsDAICase1(userTwo, tokenDai);
        paramsReceiveFixed.from = paramsPayFixed.from;
        paramsReceiveFixed.openTimestamp = paramsPayFixed.openTimestamp;

        const ItfLiquidator = await hre.ethers.getContractFactory("ItfLiquidator")
        const itfLiquidator = await ItfLiquidator.deploy(miltonDai?.address, miltonStorageDai?.address)

        await iporOracle
            .connect(userOne)
            .itfUpdateIndex(paramsPayFixed.asset, PERCENTAGE_3_18DEC, paramsPayFixed.openTimestamp);

        const volumePayFixed = 5;
        const volumeReceiveFixed = 5;

        const miltonBalanceBeforePayoutWad = USD_28_000_18DEC.mul(
            volumePayFixed + volumeReceiveFixed
        );

        await josephDai
            .connect(liquidityProvider)
            .itfProvideLiquidity(miltonBalanceBeforePayoutWad, paramsPayFixed.openTimestamp);

        const swapIdsPayFixed = [];
        const swapIdsReceiveFixed = [];

        for (let i = 0; i < volumePayFixed; i++) {
            await openSwapPayFixed(testData, paramsPayFixed);
            swapIdsPayFixed.push(i + 1);
        }

        for (let i = volumePayFixed; i < volumePayFixed + volumeReceiveFixed; i++) {
            await openSwapReceiveFixed(testData, paramsReceiveFixed);
            swapIdsReceiveFixed.push(i + 1);
        }

        const expectedSwapStatus = 0;

        const closeTimestamp = paramsPayFixed.openTimestamp.add(PERIOD_28_DAYS_IN_SECONDS);

        await miltonDai.connect(paramsPayFixed.from).itfCloseSwapPayFixed(3, closeTimestamp);
        await miltonDai.connect(paramsPayFixed.from).itfCloseSwapReceiveFixed(8, closeTimestamp);

        //when
        await itfLiquidator
            .itfLiquidate(swapIdsPayFixed, swapIdsReceiveFixed, closeTimestamp);

        //then
        for (let i = 0; i < volumePayFixed; i++) {
            const swapPayFixed = await miltonStorageDai.getSwapPayFixed(i + 1);
            expect(expectedSwapStatus, `Incorrect swap status for swapId=${i + 1}`).to.be.eq(
                swapPayFixed.state
            );
        }

        for (let i = volumePayFixed; i < volumePayFixed + volumeReceiveFixed; i++) {
            const swapReceiveFixed = await miltonStorageDai.getSwapReceiveFixed(i + 1);
            expect(expectedSwapStatus, `Incorrect swap status for swapId=${i + 1}`).to.be.eq(
                swapReceiveFixed.state
            );
        }

        const closedSwapPayFixed = await miltonStorageDai.getSwapPayFixed(3);
        expect(0, `Incorrect swap status for swapId=3`).to.be.eq(closedSwapPayFixed.state);

        const closedSwapReceiveFixed = await miltonStorageDai.getSwapReceiveFixed(8);
        expect(0, `Incorrect swap status for swapId=8`).to.be.eq(closedSwapReceiveFixed.state);
    });

```

Reply:
you can (use method itfOpenSwapPayFixed/itfOpenSwapReceiveFixed)
- open 1 swap with timestamp = z 
- opens 2. swap with timestamp = z+ 10 days

after that use itfLiquidate with timestamp+ 28 days 
only one swap should be closed

Reply:
look into this file for same examples MiltonShouldClosePositionTest